File: 2nd Year/2nd Semester/MPP/PROIECT-JAVA/Laborator/TriatlonServices/src/main/java/triatlon/services/ITriatlonServices.java
package triatlon.services;

import triatlon.model.*;

import java.util.List;
import java.util.Map;

public interface ITriatlonServices {

    // Authentification
    void login(Arbitru arbitru, ITriatlonObserver iTriatlonObserver) throws TriatlonException;

    boolean register(String username, String password, String firstName, String lastName) throws TriatlonException;

    void logout(Arbitru arbitru, ITriatlonObserver iTriatlonObserver) throws TriatlonException;


    // Rezultat Service
    void addRezultat(Participant participant, Arbitru arbitru, TipProba tipProba, int punctaj) throws TriatlonException;

    List<Rezultat> getResultateForProba(TipProba proba) throws TriatlonException;

    List<Rezultat> getAllResults();

    // Participant Service

    List<Participant> getAllParticipants() throws TriatlonException;

    int calculateTotalScore(Participant participant) throws TriatlonException;

    // Proba Service

    List<Proba> getAllProbe() throws TriatlonException;

    TipProba getProbaForArbitru(Arbitru arbitru) throws TriatlonException;

    Arbitru findArbitruByUsernameAndPassword(String username, String password);
}

File: 2nd Year/2nd Semester/MPP/Triatlon-C#/TriatlonServicess/ITriatlonServices.cs
using TriatlonModel;

namespace TriatlonServicess;

public interface ITriatlonServices
{
// Authentification
    void Login(Arbitru arbitru, ITriatlonObserver iTriatlonObserver);

    bool Register(String username, String password, String firstName, String lastName);

    void Logout(Arbitru arbitru, ITriatlonObserver iTriatlonObserver);


    // Rezultat Service
    void AddRezultat(Participant participant, Arbitru arbitru, TipProba tipProba, int punctaj);

    List<Rezultat> GetResultateForProba(TipProba proba);

    List<Rezultat> GetAllResults();

    // Participant Service

    List<Participant> GetAllParticipants();

    int CalculateTotalScore(Participant participant);

    // Proba Service

    List<Proba> GetAllProbe();

    TipProba GetProbaForArbitru(Arbitru arbitru);

    Arbitru FindArbitruByUsernameAndPassword(String username, String password);
}